src/documents/tasks.py

```diff
@@ -292,3 +296,25 @@ def update_document_archive_file(document_id):
         )
     finally:
         parser.cleanup()
+
+
+@shared_task
+def empty_trash(doc_ids=None):
+    documents = (
+        Document.deleted_objects.filter(id__in=doc_ids)
+        if doc_ids is not None
+        else Document.deleted_objects.filter(
+            deleted_at__lt=timezone.localtime(timezone.now())
+            - timedelta(
+                days=settings.EMPTY_TRASH_DELAY,
+            ),
+        )
+    )
+
+    # Temporarily connect the cleanup handler (hard_delete calls delete)
+    models.signals.post_delete.connect(cleanup_document_deletion, sender=Document)
+
+    for doc in documents:
```

Comment:
Sorry I dont exactly know what you mean by this question

Reply:
Does `documents.hard_delete()` work/exist rather than looping?

Reply:
Oh I see. Actually its sort of confusing because `Documents.objects` returns a `SoftDeleteManager` which I think has `hard_delete` but `Documents.deleted_objects` returns a `DeletedManager`, which doesnt =/

I tried playing with it a bit but maybe Im missing a (not crazy) way to do this. If its important, any thoughts?

Reply:
Ah, think I figured it out, just `delete` will be a hard delete here